### ISSUE TITLE
Is it possible to get the schema of API responses

### ISSUE THREAD

**Issue by maelle** [r-universe-org/help#542]
@jeroen if instead of only showing example output, we showed the schema of responses, could we get that schema automatically or does it need to be created manually and if so, is there a way to ensure it does not become outdated?

Idea by @llrs

**Comment by jeroen** [comment]
What sort of schema do you mean? How would you use this? Is there something comparable for the GitHub rest API that I can look at?

**Comment by maelle** [comment]
Yes see https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28 there's "response example" but also "response schema"

![image](https://github.com/user-attachments/assets/34c6b335-90a6-49b9-8056-2535fdafdb2a)

**Comment by jeroen** [comment]
Ah ok, no we don't have that, so we would have to manually write and maintain that. How would you use this?

**Comment by jeroen** [comment]
For now I don't think this is feasible, because we are still in the phase where we sometimes add or remove fields from the API. And also we don't define formal "types" for the API fields as required by schemas.

**Comment by llrs** [comment]
As a repository developer, I would like to use the API and I want to know what I can provide to users if I depend on r-universe API to enhance the R repository. If some fields are not stable this is something I would like to know. 

Documenting what is stable would be a great first step on this direction.

**Comment by jeroen** [comment]
The API is now stable. I have removed all the temporary fields that were used to store the status of some individual binaries. Instead, there is now a new `_jobs` field that holds an array with all the builds and checks that were performed.

So from here you can assume that all fields that are currently there won't be removed, and also their type will not change.

**Comment by llrs** [comment]
Thanks, nice to see 71! fields of metadata. 
One question: How to differentiate between what is on the DESCRIPTION and what is not? Initially I thought that was the "_" but then I saw fields as Repository, RemoteUrl, RemoteRef and RemoteSha (among others) that don't start with an underscore.

**Comment by jeroen** [comment]
That is a good observation. Indeed, the metadata fields prefixed with `_` do not appear in the DESCRIPTION.

The fields `Repository`, `RemoteUrl`, `RemoteRef` and `RemoteSha` get automatically inserted in the DESCRIPTION file by our build system. These same fields would be added automatically if you install a package using e.g. `install_github()`. The main purpose is that e.g. `renv` will know where to find the source for the package, if it needs to pin an exact version.

**Comment by llrs** [comment]
If this is extra fields are for compatibility with renv, would you be open to add other fields that would make it more suitable for regulated industries?
Do you mind pointing me to where this is added? For the R Regulated Repository we also add some fields but not on the DESCRIPTION field and it would interesting to see how you do this to inspire from it or adopt it. Thanks again!

**Comment by jeroen** [comment]
Sure, they get added here: https://github.com/r-universe-org/build-source/blob/d9d713db08ed2e8799b21e6b54d7415de4cce8c3/entrypoint.sh#L168-L177

**Comment by jeroen** [comment]
> would you be open to add other fields that would make it more suitable for regulated industries?

What sort of fields? Can you open a new issue for this? It is already possible to add custom metadata to your packages.json registry, which will then be copied to the r-universe API, see: https://docs.r-universe.dev/publish/set-up.html#custom-metadata
